### PR TITLE
Home page functionalities

### PR DESCRIPTION
**Pull request by abhishektanwar**
FEATURES ADDED :
- Create ,delete, update post
- Bookmark, Like post
- Comment on post
- Filter by latest or trending(most liked) posts

FILES TO CHECK
- src/components/Post/Post.jsx
- src/components/PostFilter/PostFilter.jsx
- src/components/PostFilter/post-filter.css
- src/components/Share/Share.jsx
- src/reducers/counterSlice.js

### CODE REVIEW

**Review by Ranjeetk489 — APPROVED** [review]
Good work in the coding part. UI suggestion the buttons could use some padding and your post textarea resizer need a min width property. title and favicon could be updated too. Good to merge.

File: src/components/Post/Post.jsx [hunk 1]
```diff
@@ -18,46 +65,144 @@ const Post = () => {
...
-          />
+          <p className="post-text-body body-typo-md">{desc}</p>
+          {/* {post.img ? (
+            <img
+              src={ post.img ? `data:image/png;base64,${post.img}` : dummy}
+              alt="post-image"
+              loading="lazy"
+              class="responsive-img"
+            />
+            ) : null} */}
```
Comment (pnchinmay) [inline src/components/Post/Post.jsx:116]: remove these comments

File: src/reducers/postSlice.js [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    [createNewPost.rejected]: (state, action) => {
+      state.postCreatedStatus = true;
+      state.postCreatedMessage = action.payload;
+    },
+    // delete post
+    [deletePost.pending]: (state) => {
+      state.postCreatedStatus = "loading";
+    },
+    [deletePost.fulfilled]: (state, action) => {
+      console.log("delete post", action);
```
Comment (pnchinmay) [inline src/reducers/postSlice.js:276]: remove the console statements

File: src/reducers/postSlice.js [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    },
+    [deletePost.rejected]: (state, action) => {
+      // state.postCreatedStatus = true;
+      // state.postCreatedMessage=action.payload
+    },
+    [addComment.pending]: (state) => {
+      // state.postCreatedStatus = "loading";
+    },
+    [addComment.fulfilled]: (state, action) => {
+      console.log("delete post", action);
```
Comment (pnchinmay) [inline src/reducers/postSlice.js:289]: same here

**Review by pnchinmay — APPROVED** [review]

**Pull request merged**